internal/pkg/deploy/pipeline.go

```diff
@@ -94,25 +104,77 @@ type BitbucketSource struct {
 	RepositoryURL string
 }
 
+// PipelineSourceFromManifest processes manifest info about the source based on provider type.
+// The return boolean is true for CodeStar Connections sources that require a polling prompt.
+func PipelineSourceFromManifest(source *manifest.Source) (interface{}, bool, error) {
```

Comment:
nit: you don't have to do this, it's just something i like to do :P 
When returning more than two types `(result, error)` I like to use [named return values](https://tour.golang.org/basics/7) to bring documentation into the function signature. For example:
```suggestion
func PipelineSourceFromManifest(source *manifest.Source) (source interface{}, usesCodeStar bool, err error) {
```

Reply:
Done and changed the argument name to `mfSource`.